### DISCUSSION TITLE
No internet despite connection from Wireguard

### DISCUSSION THREAD

**Discussion by milindmadhukar** [pivpn/pivpn#1644] [Support]
## In raising this issue, I confirm the following:

- [x] I have read the [documentation](https://docs.pivpn.io)
- [] Is it a feature request? please consider opening a [Discussion] (https://github.com/pivpn/pivpn/discussions/new?category=feature-requests)
- [x] Is this a support request? please consider copying the content of this document and using it to open a [Discussion](https://github.com/pivpn/pivpn/discussions/new?category=support)
- [x] I have read and understood the [PiVPN General Guidelines](https://github.com/pivpn/pivpn/#general-guidelines)
- [] The issue I am reporting can be *replicated*.  
- [] The issue I am reporting *is* directly related to the pivpn installer script.
- [x] The issue I am reporting isn't a duplicate (see [FAQs](https://github.com/pivpn/pivpn/wiki/FAQ), [closed issues](https://github.com/pivpn/pivpn/issues?q=is%3Aissue+sort%3Aupdated-desc+is%3Aclosed), and [open issues](https://github.com/pivpn/pivpn/issues?q=is%3Aissue+sort%3Aupdated-desc+is%3Aopen)).

### Has your install failed? 
```
no
```

### Describe the issue
Installation was successful but when I create a client and add it to my mobile from the Wireguard app, I see that packets are being sent but none being received and hence no internet connection.

**Expected behavior**
Need a working internet connection

**Screenshots**
![image](https://user-images.githubusercontent.com/68477234/197509468-4b59daff-7fc5-4685-a8dd-ec35f865ef81.png)

### Have you searched for similar issues and solutions?
    Yes, there are a few similar issues and all the solutions point to iptables. I have tried all of them but no avail.

**Additional context**
I think it could be something related to port forwarding but I am unsure
![image](https://user-images.githubusercontent.com/68477234/197511129-7ce34cbd-f205-4cb9-9c35-ec074cbb4d99.png)

Output of `sudo iptables -S`
```
-P INPUT ACCEPT
-P FORWARD ACCEPT
-P OUTPUT ACCEPT
-A INPUT -i eth0 -p udp -m udp --dport 51820 -m comment --comment wireguard-input-rule -j ACCEPT
-A INPUT -i wg0 -p udp -m udp --dport 53 -m comment --comment pihole-DNS-rule -j ACCEPT
-A INPUT -i wg0 -p udp -m udp --dport 53 -m comment --comment pihole-DNS-rule -j ACCEPT
```

Output of `sudo iptables -t nat -S`
```
-P PREROUTING ACCEPT
-P INPUT ACCEPT
-P OUTPUT ACCEPT
-P POSTROUTING ACCEPT
-A POSTROUTING -s 10.192.52.0/24 -o eth0 -m comment --comment wireguard-nat-rule -j MASQUERADE
-A POSTROUTING -s 10.8.0.0/24 -o eth0 -j MASQUERADE
-A POSTROUTING -s 192.168.1.0/24 -o eth0 -j MASQUERADE
```

### Have you taken any steps towards solving your issue?
Tried reinstalling, looking at debug logs and finding solutions over the interent.

#### What type of hardware are you running PiVPN at?
Raspberry Pi 4B (4GB Ram)

#### Output of `uname -a`
```
Linux raspberrypi 5.15.61-v8+ #1579 SMP PREEMPT Fri Aug 26 11:16:44 BST 2022 aarch64 GNU/Linux
```

#### Output of `cat /etc/os-release`
```
PRETTY_NAME="Debian GNU/Linux 11 (bullseye)"
NAME="Debian GNU/Linux"
VERSION_ID="11"
VERSION="11 (bullseye)"
VERSION_CODENAME=bullseye
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
```

### Console output of      `curl -L install.pivpn.io | bash`
![image](https://user-images.githubusercontent.com/68477234/197511777-b74bf457-8504-4a43-93e8-d2c1386aaace.png)

### Console output of      `pivpn add` or `pivpn add nopass`
```
Enter a Name for the Client: testclient
::: Client Keys generated
::: Client config generated
::: Updated server config
::: WireGuard reloaded
======================================================================
::: Done! testclient.conf successfully created!
::: testclient.conf was copied to /home/pi/configs for easytransfer.
::: Please use this profile only on one device and create additional
::: profiles for other devices. You can also use pivpn -qr
::: to generate a QR Code you can scan with the mobile app.
======================================================================
```

### Console output of      `pivpn debug`

```
::: Generating Debug Output
::::		PiVPN debug		 ::::
=============================================
::::		Latest commit		 ::::
Branch: master
Commit: ab1fe203992dc0ef586644efb12488b876ed9555
Author: 4s3ti
Date: Sat Oct 8 02:39:53 2022 +0200
Summary: fix(ufw): set missing openvpn ipv6 variables
=============================================
::::	    Installation settings    	 ::::
PLAT=Debian
OSCN=bullseye
USING_UFW=0
IPv4dev=eth0
IPv6dev=eth0
IPv4addr=192.168.1.96/24
IPv4gw=192.168.1.1
install_user=pi
install_home=/home/pi
VPN=wireguard
pivpnPORT=51820
pivpnDNS1=8.8.8.8
pivpnDNS2=8.8.4.4
pivpnHOST=REDACTED
INPUT_CHAIN_EDITED=1
FORWARD_CHAIN_EDITED=0
INPUT_CHAIN_EDITEDv6=0
FORWARD_CHAIN_EDITEDv6=0
pivpnPROTO=udp
pivpnMTU=1420
pivpnDEV=wg0
pivpnNET=10.192.52.0
subnetClass=24
pivpnenableipv6=1
pivpnNETv6="fd11:5ee:bad:c0de::"
subnetClassv6=64
ALLOWED_IPS="0.0.0.0/0, ::0/0"
UNATTUPG=1
INSTALLED_PACKAGES=()
=============================================
::::  Server configuration shown below   ::::
[Interface]
PrivateKey = server_priv
Address = 10.192.52.1/24,fd11:5ee:bad:c0de::1/64
MTU = 1420
ListenPort = 51820
### begin testclient ###
[Peer]
PublicKey = testclient_pub
PresharedKey = testclient_psk
AllowedIPs = 10.192.52.3/32,fd11:5ee:bad:c0de::3/128
### end testclient ###
=============================================
::::  Client configuration shown below   ::::
[Interface]
PrivateKey = testclient_priv
Address = 10.192.52.3/24,fd11:5ee:bad:c0de::3/64
DNS = 8.8.8.8, 8.8.4.4

[Peer]
PublicKey = server_pub
PresharedKey = testclient_psk
Endpoint = REDACTED:51820
AllowedIPs = 0.0.0.0/0, ::0/0
=============================================
:::: 	Recursive list of files in	 ::::
::::	/etc/wireguard shown below	 ::::
/etc/wireguard:
configs
keys
wg0.conf

/etc/wireguard/configs:
clients.txt
testclient.conf

/etc/wireguard/keys:
server_priv
server_pub
testclient_priv
testclient_psk
testclient_pub
=============================================
::::		Self check		 ::::
:: [OK] IP forwarding is enabled
:: [OK] Iptables MASQUERADE rule set
:: [OK] Iptables INPUT rule set
:: [OK] WireGuard is running
:: [OK] WireGuard is enabled 
(it will automatically start on reboot)
:: [OK] WireGuard is listening on port 51820/udp
=============================================
:::: Having trouble connecting? Take a look at the FAQ:
:::: https://docs.pivpn.io/faq
=============================================
:::: WARNING: This script should have automatically masked sensitive       ::::
:::: information, however, still make sure that PrivateKey, PublicKey      ::::
:::: and PresharedKey are masked before reporting an issue. An example key ::::
:::: that you should NOT see in this log looks like this:                  ::::
:::: YIAoJVsdIeyvXfGGDDadHh6AxsMRymZTnnzZoAb9cxRe                          ::::
=============================================
::::		Debug complete		 ::::
::: 
::: Debug output completed above.
::: Copy saved to /tmp/debug.log
::: 
```

**Comment by coolapso** [comment]
> Yes, there are a few similar issues and all the solutions point to iptables. I have tried all of them but no avail. 

Not all solutions related with this problem are related to IPTables.  and the most common is trying to connect while using the same network as the vpn server. this being said: 

* Are you trying to connect within the same lan as the Wireguard server? IE. Is your Wireguard server at home and trying to connect while at home? If so please try connecting from another network, example, mobile network, starbucks, any network other than the same where the Wireguard server is.

**Reply by milindmadhukar** [reply]
I have tried using a different network (mobile data) and my same local network, both of them seem to have the same problem, packets are not received.
Funny thing is I had set this up about half an year back and it worked like a charm but now something seems to have broken

**Comment by milindmadhukar** [comment]
![image](https://user-images.githubusercontent.com/68477234/197760285-153a79b6-0704-4f3f-a078-96e0b376d959.png)
Looking at the wireguard logs, this is what I see. Idk if it helps but its something

**Comment by thewriteway** [comment]
two things I can think of:

1) Check your public listen port is still active and being forwarded to your pivpn.
2) Set the client and the VPN's own DNS servers to 10.192.52.1 (so it forces the VPN and client to route DNS lookups internally) 

   i.e. changing these values below to 10.192.52.1
   pivpnDNS1=8.8.8.8
   pivpnDNS2=8.8.4.4
   DNS = 8.8.8.8, 8.8.4.4

That has worked for mine.
